core/src/test/java/org/bitcoinj/core/CoinTest.java

```diff
@@ -80,6 +84,54 @@ public void testValueOf() {
         } catch (IllegalArgumentException e) {}
     }
 
+
+    @Test
+    public void testbtcToSatoshi() {
+        assertEquals(Long.MIN_VALUE, btcToSatoshi(new BigDecimal("-92233720368.54775808")));
+        assertEquals(0L, btcToSatoshi(BigDecimal.ZERO));
+        assertEquals(COIN.value, btcToSatoshi(BigDecimal.ONE));
+        assertEquals(Long.MAX_VALUE, btcToSatoshi(new BigDecimal("92233720368.54775807")));
+    }
+
+    @Test(expected = ArithmeticException.class)
+    public void testofBtcTooSmall() {
+        btcToSatoshi(new BigDecimal("-92233720368.54775809"));
```

Comment:
Perhaps add a comment like 'a bit smaller than Long.MIN_VALUE'?

Reply:
Fixed in 165341fbf4a027f9375561964b52a0e88f9073d1